lib/model/product.rb

```diff
@@ -5,13 +5,12 @@ class Product
     attr_reader :name, :id
     attr_accessor :price, :vat, :quantity
 
-    @@id = 0
     def initialize(id:, name:, price:, vat:, quantity:)
       @id = id.to_i
-      @name = set_name(name)
-      @price = set_price(price)
-      @vat = set_vat(vat)
-      @quantity = set_quantity(quantity)
+      @name = validate_name(name)
+      @price = validate_price(price)
+      @vat = validate_vat(vat)
+      @quantity = validate_quantity(quantity)
```

Comment:
This is a good step in naming, but imho validation should not be done on creating new object, but before saving to database.